src/ivc/cyclefold/support_circuit/mod.rs

```diff
@@ -0,0 +1,161 @@
+    main_gate::RegionCtx,
+};
+
+type EccChip<C> = ecc::EccChip<C, tiny_gate::Gate<<C as CurveAffine>::Base>>;
+
+mod tiny_gate;
+use tiny_gate::{Config as GateConfig, Gate};
+
+#[derive(Default)]
+pub struct SupportCircuit<C: CurveAffine> {
+    _p: PhantomData<C>,
+}
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    gate_config: GateConfig,
+    instance: Column<Instance>,
+}
+
+impl<C: CurveAffine> Circuit<C::Base> for SupportCircuit<C>
+where
+    C::Base: PrimeFieldBits,
+{
+    type Config = Config;
+    type FloorPlanner = SimpleFloorPlanner;
+
+    fn without_witnesses(&self) -> Self {
+        SupportCircuit { _p: PhantomData }
+    }
+
+    fn configure(meta: &mut ConstraintSystem<C::Base>) -> Self::Config {
+        let instance = meta.instance_column();
+        meta.enable_equality(instance);
+
+        Config {
+            gate_config: Gate::configure(meta),
+            instance,
+        }
+    }
+
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<C::Base>,
+    ) -> Result<(), Halo2PlonkError> {
+        let ecc_chip = EccChip::<C>::new(config.gate_config.clone());
+
+        layouter.assign_region(
+            || "",
+            |region| {
+                let mut ctx = RegionCtx::new(region, 0);
+
+                let [x0, y0, l0, x1, y1, l1, _expected_x, _expected_y] = ecc_chip
+                    .gate
+                    .assign_values_from_instance(&mut ctx, config.instance, 0)
+                    .unwrap();
+
+                trace!("instances assigned");
+
+                let [p0, p1] = [
+                    AssignedPoint::<C> { x: x0, y: y0 },
+                    AssignedPoint::<C> { x: x1, y: y1 },
+                ];
+
+                let num_bits =
+                    NonZeroUsize::new(<C::Base as PrimeField>::NUM_BITS as usize).unwrap();
+
+                let l0_bits = ecc_chip
+                    .gate
+                    .le_num_to_bits(&mut ctx, &l0, num_bits)
+                    .unwrap();
+                let lhs = ecc_chip
+                    .scalar_mul_non_zero(&mut ctx, &p0, &l0_bits)
+                    .unwrap();
+                trace!("p0 * l0_bits = [{:?},{:?}]", lhs.x.value(), lhs.y.value());
+
+                let l1_bits = ecc_chip
+                    .gate
+                    .le_num_to_bits(&mut ctx, &l1, num_bits)
+                    .unwrap();
+                trace!("l1 bits ready");
+                let rhs = ecc_chip
+                    .scalar_mul_non_zero(&mut ctx, &p1, &l1_bits)
+                    .unwrap();
+                trace!("p1 * l1_bits");
+
+                let AssignedPoint {
+                    x: _actual_x,
+                    y: _actual_y,
+                } = ecc_chip.add(&mut ctx, &lhs, &rhs).unwrap();
+                trace!("add finished");
+
+                //ctx.constrain_equal(expected_x.cell(), actual_x.cell())
+                //    .unwrap();
```

Comment:
what does the "actual_x" suppose to be? I think for support circuit, we don't need to do any consistency checking.

Reply:
This will allow us to just take the data from the instance column for the primary. I've just removed the constraint equal for now, since the {off,on}-circuit consistency is not trivially checked (it's in the ecc code, but I've put off checking it)

And of course we need it, we eval off-cictuit via protogalaxy and it's must be same

Reply:
The support circuit doesn't worry about the instance from primary, since the primary circuit will check the input consistency.
The support circuit simply do ecc mul whatever the input is.

Reply:
What I meant was that within sfc we need to take the result of this operation

Reply:
IC, so this comment is just a reminder?

Reply:
No, the test doesn't work with this comment, because scalar multiplication works differently for some reason (and there is a whole separate class in the ecc gadget tests that implements it). This will become a problem when we start testing circuit <-> nifs consistency. I left this code to test this logic soon to see what the difference is.

I don't usually do this, but I'm in a hurry here

Reply:
understood for the testing purpose.